### DISCUSSION TITLE
Using the glossary to answer questions

### DISCUSSION THREAD

**Discussion by jsvgoncalves** [Depart-de-Sentier/sentier.dev#28] [Workshop Item]
**Goal:** Explore how to combine the power of large language models (LLMs) with knowledge graphs derived from the LCA glossary to provide accurate and context-aware answers to LCA-related questions.

**Key Questions:**
- **Knowledge Graph Representation:** Is the current data model sufficiently fit for purpose or do need to move to e.g., using RDF, Neo4j?
- **RAG Integration:** What are the best practices for integrating LLMs with knowledge graphs (e.g., using frameworks like LangChain)?

**Potential Outcomes:**
- A prototype question-answering system that demonstrates the potential of RAG for LCA.
- A roadmap for further development and deployment of this system.

**Comment by tngTUDOR** [comment]
What is the RAG ?

**Reply by jsvgoncalves** [reply]
Here's a good source https://neo4j.com/developer-blog/fine-tuning-retrieval-augmented-generation/

**Comment by simontaurus** [comment]
We made some first experiences with reconstructing simple human language via SPARQL CONSTRUCT, serializing triples to short
"subject_label property_label object_label"
sentences, e. g. "9SMn28 is a subclass of Steel".
Latest LLMs like GPT4 are also able to generate SPARQL from text
